frontend/rust-lib/flowy-search/src/folder/indexer.rs

```diff
@@ -0,0 +1,223 @@
+    let searcher = reader.searcher();
+
+    let top_docs: Vec<(Score, DocAddress)> = searcher
+      .search(&built_query, &TopDocs::with_limit(10))
+      .unwrap();
+
+    let mut search_results: Vec<SearchResultPB> = vec![];
+    for (_score, doc_address) in top_docs {
+      let retrieved_doc = searcher.doc(doc_address).unwrap();
+
+      let mut content = HashMap::new();
+      let named_doc = self.folder_schema.schema.to_named_doc(&retrieved_doc);
+      for (k, v) in named_doc.0 {
+        content.insert(k, v[0].clone());
+      }
+
+      if content.len() == 0 {
+        continue;
+      }
+
+      let s = serde_json::to_string(&content)?;
+      let result: SearchResultPB = serde_json::from_str::<FolderIndexData>(&s).unwrap().into();
+      search_results.push(result);
+    }
+
+    Ok(search_results)
+  }
+}
+
+impl IndexManager for FolderIndexManager {
+  fn set_index_content_receiver(&self, mut rx: IndexContentReceiver) {
+    let indexer = self.clone();
+    af_spawn(async move {
+      while let Ok(msg) = rx.recv().await {
+        match msg {
+          IndexContent::Create(value) => match serde_json::from_value::<ViewIndexContent>(value) {
+            Ok(view) => {
+              let _ = indexer.add_index(IndexableData {
+                id: view.id,
+                data: view.name,
+              });
+            },
+            Err(err) => tracing::error!("FolderIndexManager error deserialize: {:?}", err),
+          },
+          IndexContent::Update(value) => match serde_json::from_value::<ViewIndexContent>(value) {
+            Ok(view) => {
+              let _ = indexer.update_index(IndexableData {
+                id: view.id,
+                data: view.name,
+              });
+            },
+            Err(err) => tracing::error!("FolderIndexManager error deserialize: {:?}", err),
+          },
+          IndexContent::Delete(ids) => {
+            if let Err(e) = indexer.remove_indices(ids) {
+              tracing::error!("FolderIndexManager error deserialize: {:?}", e);
+            }
+          },
+        }
+      }
+    });
+  }
+
+  fn update_index(&self, data: IndexableData) -> Result<(), FlowyError> {
+    let mut index_writer = self.index.writer(50000000).unwrap();
+
+    let id_field = self
+      .folder_schema
+      .schema
+      .get_field(FOLDER_ID_FIELD_NAME)
+      .unwrap();
+    let title_field = self
+      .folder_schema
+      .schema
+      .get_field(FOLDER_TITLE_FIELD_NAME)
+      .unwrap();
+
+    let delete_term = Term::from_field_text(id_field, &data.id.clone());
+
+    // Remove old index
+    index_writer.delete_term(delete_term);
+
+    // Add new index
+    let _ = index_writer.add_document(doc![
+      id_field => data.id.clone(),
+      title_field => data.data,
+    ]);
+
+    index_writer.commit()?;
+
+    Ok(())
+  }
+
+  fn remove_indices(&self, ids: Vec<String>) -> Result<(), FlowyError> {
+    let mut index_writer = self.index.writer(50000000).unwrap();
```

Comment:
what is 50000000 for?

Reply:
It's the allocated heap in bytes (50MB), I'll add a comment.